src/auth/components/register/RegisterForm.tsx

```diff
@@ -0,0 +1,348 @@
+
+import { connect } from "react-redux";
+
+import { ThunkDispatch } from "redux-thunk";
+
+import { AnyAction } from "redux";
+
+import { sendOTP } from "../../core/services/register";
+
+import { RegisterUser, SetAuthenticated } from "../../core/redux/actions";
+
+import { StateType } from "../../../app/core/redux/types";
+
+type AuthProps = {
+  isAuthenticated: boolean;
+  error?: string;
+  success?: boolean;
+  message?: string;
+};
+
+interface RegisterData {
+  first_name: string;
+  last_name: string;
+  email: string;
+  password: string;
+  phone_number: string;
+  phone_number_extension: string;
+  otp: string;
+}
+
+type Props = {
+  setAuthenticated: (value?: boolean) => void;
+  registerUser: (registerData: RegisterData) => void;
+  auth: AuthProps;
+};
+
+const RegisterForm: React.FC<Props> = ({
+  setAuthenticated,
+  registerUser,
+  auth,
+}) => {
+  const [isOTPSent, setIsOTPSent] = useState(false);
+  const [isResendAllowed, setIsResendAllowed] = useState(false);
+
+  let counter = 60;
+
+  let otpCountdown: any;
+
+  const countdown = () => {};
+
+  const setOTPTimeout = () => {
+    otpCountdown = setInterval(() => {
+      if (counter > 0) {
+        const newCounter = counter - 1;
+        counter = counter - 1;
+        console.log("counter value : ", newCounter);
+      } else {
+        counter = 11;
+        setIsResendAllowed(true);
+        clearInterval(otpCountdown);
+        return;
+      }
+    }, 1000);
+  };
+
+  const handleGetOTP = (phone: string) => {
+    setIsOTPSent(true);
+    setIsResendAllowed(false);
+    setOTPTimeout();
+    sendOTPFunc(phone);
+  };
+
+  const sendOTPFunc = (phone: string) => {
+    isOTPSent ? handleResendOTP(phone) : sendOTP(phone);
+  };
+
+  const handleResendOTP = (phone: string) => {
+    countdown();
+  };
+
+  const redirectToDashboard = () => {
+    setTimeout(() => {
+      setAuthenticated(true);
+    }, 1000);
+  };
+
+  return (
+    <div>
+      <Card>
+        <CardHeader title="Registration Form"></CardHeader>
+        <CardContent>
+          {auth.hasOwnProperty("error") && (
+            <Typography color="error" component="h5" variant="h5">
+              {auth.error}
+            </Typography>
+          )}
+          {auth.hasOwnProperty("success") ? (
+            <Typography color="primary" component="h5" variant="h5">
+              {auth.success}
+              Redirecting to dashboard.
+              {redirectToDashboard()}
+            </Typography>
+          ) : (
+            <Formik
+              initialValues={{
+                firstName: "",
+                lastName: "",
+                email: "",
+                password: "",
+                password2: "",
+                phone: "",
+                otp: "",
+                phone_number_extension: "+91",
+              }}
+              validate={(values) => {
+                const errors: any = {};
+                if (!values.firstName) {
```

Comment:
extract validations to a different method

Reply:
Done.